syntax = "proto3";

package services;

service DoStuff {
    rpc say_hello (HelloRequest) returns (HelloReply) {}
    rpc say_hello_again (HelloRequest) returns (HelloReply) {}
    rpc get_service_port (ServiceName) returns (ServicePort) {}
    rpc get_service_description (ServiceName) returns (ServiceDescription) {}
}

message HelloRequest {
    int32 pid = 1;
}

message HelloReply {
    string retval = 1;
}

message ServiceName {
    string name = 1;
}

message ServicePort {
    int32 port = 1;
}

message ServiceDescription {
    string description = 1;
}